drivers/hmis/app/models/hmis/file.rb

```diff
@@ -25,9 +25,10 @@ def data_source
   belongs_to :enrollment, class_name: '::Hmis::Hud::Enrollment', optional: true
   belongs_to :client, class_name: '::Hmis::Hud::Client'
   belongs_to :user, class_name: 'Hmis::Hud::User', optional: true
+  belongs_to :uploaded_by, class_name: 'Hmis::Hud::User', optional: true
```

Comment:
I think these should both be `class_name: 'Hmis::User'`

Reply:
Yeah. I must have missed this in the initial review @ZachCMP. There is inconsistency - some places we expect the `user_id` to point to a HUD user and other places we expect an HMIS user. It should be the user ID of the Hmis::User aka the "application user." I just left some comments to point out that places that used HUD User on the original PR: https://github.com/greenriver/hmis-warehouse/pull/2901#pullrequestreview-1371515726

@eanders correct me if I'm wrong, but in the warehouse the `user_id` column stores the user that _uploaded_ the file. We do not currently store the user who most recently _updated_ it.

My suggestion to @ZachCMP  was that we continue to not store the updated_by information for now, since we haven't needed it so far in the warehouse. But if we DO want to store it, it should be a new column call `updated_by` (also a User, not a HUD User) so that we can continue to treat the `user_id` column as its current function – the user who uploaded. Again please confirm this is how we use it in the warehouse.

Reply:
That's probably correct, I didn't catch that.  `user` is probably the creator, so if we want to know who touched it most-recently, we'd need to track that in `updated_by`.

Reply:
Ok. Just confirmed, the warehouse does not have the ability to edit File records. So, the `user_id` currently stores the "uploader" and we should keep it that way.

Reply:
@gigxz @eanders Maybe I'm misunderstanding, but it looks to me like this block from the submit form mutation updates the user with the `Hmis::Hud::User` every time an object is created or updated. Am I missing something here?

```ruby
# Find or create record
      hud_user = Hmis::Hud::User.from_user(current_user)
      if input.record_id.present?
        record = klass.viewable_by(current_user).find_by(id: input.record_id)
        record&.assign_attributes(user: hud_user)
      else
        record = klass.new(
          user: hud_user,
          data_source_id: hud_user.data_source_id,
          **related_id_attributes(klass.name, input),
        )
      end
```

Reply:
@ZachCMP that's probably correct.  For the classes that are "HUD" classes, UserID should be the Hmis::Hud::User's ID.  For things that don't fall within that `Hmis::Hud` namespace, `user_id` should reverence `Hmis::User`.